keras/src/models/model.py

```diff
@@ -556,6 +556,125 @@ def _get_variable_map(self):
         map_saveable_variables(self, store=store, visited_saveables=set())
         return store
 
+    def get_nested_variables(
+        self,
+        trainable_variables=False,
```

Comment:
Do we actually need these args? Maybe not.

Reply:
I was trying to match [`_get_jax_state` signature](https://github.com/keras-team/keras/blob/master/keras/src/backend/jax/trainer.py#L943-L948).
So you're saying there isn't any use case that users want to get some of the variables only?

Reply:
`_get_jax_state()` is private and very adhoc. We don't need these args.

Reply:
Should we always include metrics and optimizer variables then?

Reply:
Yes?